tests/test_codebase/test_mmdet3d/test_monocular_detection.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+from tempfile import NamedTemporaryFile, TemporaryDirectory
+
+import mmcv
+import pytest
+import torch
+from torch.utils.data import DataLoader
+from torch.utils.data.dataset import Dataset
+
+import mmdeploy.backend.onnxruntime as ort_apis
+from mmdeploy.apis import build_task_processor
+from mmdeploy.codebase import import_codebase
+from mmdeploy.utils import Codebase, load_config
+from mmdeploy.utils.test import DummyModel, SwitchBackendWrapper
+
+try:
+    import_codebase(Codebase.MMDET3D)
+except ImportError:
+    pytest.skip(
+        f'{Codebase.MMDET3D} is not installed.', allow_module_level=True)
+
+model_cfg_path = 'tests/test_codebase/test_mmdet3d/data/monodet_model_cfg.py'
+img_path = 'tests/test_codebase/test_mmdet3d/data/nuscenes/' \
+           'n015-2018-07-24-11-22-45+0800__CAM_BACK__1532402927637525.jpg'
+model_cfg = load_config(model_cfg_path)[0]
+deploy_cfg = mmcv.Config(
+    dict(
+        backend_config=dict(type='onnxruntime'),
+        codebase_config=dict(
+            type='mmdet3d',
+            task='MonocularDetection',
+            ann_file='tests/test_codebase/test_mmdet3d/data/nuscenes/n015-2018'
+            '-07-24-11-22-45+0800__CAM_BACK__1532402927637525_mono3d.coco.json'
+        ),
+        onnx_config=dict(
+            type='onnx',
+            export_params=True,
+            keep_initializers_as_inputs=False,
+            opset_version=11,
+            input_shape=None,
+            input_names=['img', 'cam2img', 'cam2img_inverse'],
+            output_names=['bboxes', 'scores', 'labels', 'dir_scores',
+                          'attrs'])))
+onnx_file = NamedTemporaryFile(suffix='.onnx').name
+task_processor = build_task_processor(model_cfg, deploy_cfg, 'cpu')
+num_classes = 10
+num_attr = 5
+num_dets = 20
+
+
+def test_init_pytorch_model():
```

Comment:
Test failed with log:

```dotnetcli
__________________________________________________________________ test_init_pytorch_model ___________________________________________________________________
                                                                                            
    def test_init_pytorch_model():                                                                                                                            
        from mmdet3d.models import SingleStageMono3DDetector                         
>       model = task_processor.init_pytorch_model(None)

tests/test_codebase/test_mmdet3d/test_monocular_detection.py:54:                           
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mmdeploy/codebase/mmdet3d/deploy/monocular_detection.py:63: in init_pytorch_model                                                                             
    model = init_model(self.model_cfg, model_checkpoint, device)                                                                                    
../mmdetection3d/mmdet3d/apis/inference.py:60: in init_model                                                               
    model = build_model(config.model, test_cfg=config.get('test_cfg'))
../mmdetection3d/mmdet3d/models/builder.py:119: in build_model
    if cfg.type in ['EncoderDecoder3D']:                
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                                                    
self = {'backbone': {'type': 'ResNet', 'depth': 101, 'num_stages': 4, 'out_indices': (0, 1, 2, 3), 'frozen_stages': 1, 'norm_...False, 'nms_pre': 1000, 'nms_t
hr': 0.8, 'score_thr': 0.05, 'min_bbox_size': 0, 'max_per_img': 200}, 'pretrained': None}
name = 'type'

    def __getattr__(self, name):
        try:
            value = super().__getattr__(name)
        except KeyError:
            ex = AttributeError(f"'{self.__class__.__name__}' object has no "
                                f"attribute '{name}'")
        except Exception as e:
            ex = e
        else:
            return value
>       raise ex
E       AttributeError: 'ConfigDict' object has no attribute 'type'

../mmcv/mmcv/utils/config.py:50: AttributeError

```

Reply:
test_run_inference, test_visualize failed with same error.

Reply:
fix it in latest commit.